scipy/io/tests/test_mmio.py

```diff
@@ -28,6 +28,22 @@ def test_simple(self):
         b = mmread(fn)
         assert_array_almost_equal(a,b)
 
+    def test_simple_upper_triangle(self):
+        a = [[0,1],[0,0]]
```

Comment:
There is also the test `test_random_symmetric_real` for a bigger random symmetric matrix once as (dense) array in `TestMMIOArray` and once as sparse matrix in `TestMMIOSparseCSR`. I think it is unnecessary to test more sizes.